### ISSUE TITLE
StackOverflowError exception when mapping enum

### ISSUE THREAD

**Issue by Kiramind** [orika-mapper/orika#166]
The following simple test works perfectly in orika 1.4.6 and throws `StackOverflowError` in 1.5.0

Am I missing some new usage ?

```java
import org.junit.Assert;
import org.junit.Test;

import ma.glasnost.orika.MapperFacade;
import ma.glasnost.orika.MapperFactory;
import ma.glasnost.orika.impl.DefaultMapperFactory;

public class SimpleBeanTest {

    @Test
    public void test() throws Exception {

        MapperFactory factory = new DefaultMapperFactory.Builder().build();
        MapperFacade beanMapper = factory.getMapperFacade();

        SimpleBeanResource sbr = beanMapper.map(SimpleBean.E1, SimpleBeanResource.class);
        Assert.assertTrue(sbr.getName().equals(SimpleBean.E1.getName()));
    }
}
```
```java
import java.io.Serializable;

public class SimpleBeanResource implements Serializable {

    private String code;
    private String name;

    public SimpleBeanResource(String code, String name) {
        this.code = code;
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
```
```java
public enum SimpleBean {
    E1("code_e1", "name_e1");

    SimpleBean(String code, String name) {
        this.code = code;
        this.name = name;
    }

    private final String code;

    private final String name;

    public String getCode() {
        return code;
    }

    public String getName() {
        return name;
    }
}
```
# Thrown exception : 
```
java.lang.StackOverflowError
	at java.lang.Class.getEnclosingMethodInfo(Class.java:1072)
	at java.lang.Class.getEnclosingClass(Class.java:1272)
	at java.lang.Class.getSimpleBinaryName(Class.java:1443)
	at java.lang.Class.getSimpleName(Class.java:1309)
	at java.lang.Class.isAnonymousClass(Class.java:1411)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:374)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:361)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:295)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:321)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:299)
	at ma.glasnost.orika.metadata.TypeUtil.convertTypeArguments(TypeUtil.java:193)
	at ma.glasnost.orika.metadata.TypeFactory.intern(TypeFactory.java:420)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:378)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:361)
	at ma.glasnost.orika.metadata.TypeFactory.limitedValueOf(TypeFactory.java:295)
...
```

**Comment by andreabertagnolli** [comment]
The problem was born after the resolution of #67 , I removed two lines that added and removed the current rawType to the recursive bounds ( before and after line 193 https://github.com/orika-mapper/orika/blob/master/core/src/main/java/ma/glasnost/orika/metadata/TypeUtil.java#L193 ).

That patch fixed #67 but break this.
I'll take a look soon, in the meantime someone's got other ideas?

**Comment by brabenetz** [comment]
I'm on it and will provide a patch.
